test/e2e/node/node_problem_detector.go

```diff
@@ -53,19 +53,17 @@ var _ = SIGDescribe("NodeProblemDetector", func() {
 
 	ginkgo.BeforeEach(func(ctx context.Context) {
 		e2eskipper.SkipUnlessSSHKeyPresent()
-		e2eskipper.SkipUnlessProviderIs(framework.ProvidersWithSSH...)
 		e2eskipper.SkipUnlessProviderIs("gce", "gke")
-		e2eskipper.SkipUnlessNodeOSDistroIs("gci", "ubuntu")
 		e2enode.WaitForTotalHealthy(ctx, f.ClientSet, time.Minute)
 	})
+	hostExec := utils.NewHostExec(f)
 
 	ginkgo.It("should run without error", func(ctx context.Context) {
 		ginkgo.By("Getting all nodes and their SSH-able IP addresses")
 		readyNodes, err := e2enode.GetReadySchedulableNodes(ctx, f.ClientSet)
 		framework.ExpectNoError(err)
 
 		nodes := []v1.Node{}
-		hosts := []string{}
```

Comment:
So we will never need this fallback to the IP address any longer? I don't have all context on why it was needed. Can you please elaborate a bit in the description?

Reply:
We don't need IP addresses anymore. I can delete the for loop below.

Reply:
Fixed

Reply:
The scale test might need this logic that I deleted.

> test fails with
> 
> > No node-problem-detector pod is present in


The pod is definitely being deployed. Taking a look